### DISCUSSION TITLE
Why isn't this workflow with both `on.push.paths` and `on.push.tags` running?

### DISCUSSION THREAD

**Discussion by SyntaxColoring** [community/community#26821] [Actions]
<p>We have a workflow that we want to run whenever <em>either</em> of the following are true:</p>
<ol>
<li>A push, to any branch, modifies certain paths.</li>
<li>A tag is pushed that looks like <code>docs@foo</code>.</li>
</ol>
<p>Here is what we have so far:</p>
<aside class="onebox githubblob" data-onebox-src="https://github.com/Opentrons/opentrons/blob/26f8ea431ca75881c71a24d887b85daae1d4738c/.github/workflows/docs-build.yaml#L7-L18">
  <header class="source">

      <a href="https://github.com/Opentrons/opentrons/blob/26f8ea431ca75881c71a24d887b85daae1d4738c/.github/workflows/docs-build.yaml#L7-L18" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/Opentrons/opentrons/blob/26f8ea431ca75881c71a24d887b85daae1d4738c/.github/workflows/docs-build.yaml#L7-L18" target="_blank" rel="noopener nofollow ugc">Opentrons/opentrons/blob/26f8ea431ca75881c71a24d887b85daae1d4738c/.github/workflows/docs-build.yaml#L7-L18</a></h4>

  <pre class="onebox">    <code class="lang-yaml">
      <ol class="start lines" start="7" style="counter-reset: li-counter 6 ;">
          <li>on:</li>
          <li>  push:</li>
          <li>    paths:</li>
          <li>      - 'api/src/opentrons/protocol_api/**'</li>
          <li>      - 'api/docs/**'</li>
          <li>      - '.github/workflows/docs-build.yaml'</li>
          <li>      - '.github/actions/python/**'</li>
          <li>      - '.github/actions/webstack/deploy-to-sandbox/**'</li>
          <li>      - '.github/workflows/utils.js'</li>
          <li>    tags:</li>
          <li>      - 'docs@*'</li>
          <li>  workflow_dispatch:</li>
      </ol>
    </code>
  </pre>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Unexpectedly, this does not seem to run at all when a commit modifies the designated paths but <em>isn’t</em> tagged with <code>docs@foo</code>. <a href="https://github.com/Opentrons/opentrons/pull/8452" rel="noopener nofollow ugc">Here’s an example PR</a> that I expect to trigger the workflow, but doesn’t.</p>
<p>What am I missing? I expect <code>on.push.paths</code> and <code>on.push.tags</code> to be evaluated with a logical <code>OR</code>. Is it logical <code>AND</code> instead, or something?</p>

**Comment by kingthorin** [comment]
<p>Push != PR</p>
<p>This will only run for pushes that modify something in the designated paths and are tagged as expected.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>See <a href="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#onpushpull_requestbranchestags">the documentation</a>:</p>
<blockquote>
<p>If you define only <code>tags</code> or only <code>branches</code> , the workflow won’t run for events affecting the undefined Git ref.</p>
</blockquote>
<p>So if you want the workflow to run for any branch in addition to the matching tags, add something like</p>
<pre><code class="lang-yaml">    branches:    
      - '**'
</code></pre>
<p>to the the <code>push</code> conditions. Note that either way the workflow will only run if the path filter matches.</p>
